opentelemetry-sdk/src/trace/span_processor_with_async_runtime.rs

```diff
@@ -233,17 +234,24 @@ impl<E: SpanExporter, R: RuntimeChannel> BatchSpanProcessorInternal<E, R> {
         match message {
             // Span has finished, add to buffer of pending spans.
             BatchMessage::ExportSpan(span) => {
-                self.spans.push(span);
+                if self.spans.len() == self.config.max_export_batch_size {
+                    // Replace the oldest span with the new span to avoid suspending messages
+                    // processing.
+                    self.spans.pop_front();
```

Comment:
This seems to be silently dropping the old span - Should we log a warning here?

Reply:
@lalitb We should. Also, we probably need to bound on `self.config.max_queue_size` instead of `self.config.max_export_batch_size`. `BatchConfig` notifies the spans will be dropped only if `max_queue_size` reached, not `max_export_batch_size`.